front/app/containers/Admin/projects/project/analysis/TopBar/Filters/AuthorFilters.tsx

```diff
@@ -49,42 +55,102 @@ const AuthorFilters = () => {
     []
   );
 
+  const selectedGenderOptions = handleArraySearchParam(
+    searchParams,
+    genderUrlQueryParamKey
+  );
+
+  const selectedDomicileOptions = handleArraySearchParam(
+    searchParams,
+    domicileUrlQueryParamKey
+  );
+
+  const toggleOptionInArray = (array: string[] | undefined, option: string) => {
+    if (array?.includes(option)) {
+      const arrayWithoutOption = array.filter((el: string) => el !== option);
+      return arrayWithoutOption.length > 0 ? arrayWithoutOption : undefined;
+    }
+    return array ? [...array, option] : [option];
+  };
```

Comment:
I think you could shorten this to lodash `xor(array, [option])`, you might just need to still deal with the `undefined` when empty thing.

Reply:
I think this is pretty explicit and clear, not sure we need an additional helper for it.